x-pack/plugins/infra/public/components/asset_details/tabs/overview/kpis/kpi_grid.tsx

```diff
@@ -4,21 +4,55 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-import React from 'react';
+
+import React, { useMemo } from 'react';
 import { EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
-import { Tile, type TileProps } from './tile';
-import { KPI_CHARTS } from '../../../../../common/visualizations/lens/dashboards/host/kpi_grid_config';
+import type { DataView } from '@kbn/data-views-plugin/public';
+import type { TimeRange } from '@kbn/es-query';
+import { LensChart, TooltipContent } from '../../../../lens';
+import { buildCombinedHostsFilter } from '../../../../../utils/filters/build';
+import {
+  KPI_CHARTS,
+  KPI_CHART_HEIGHT,
+  AVERAGE_SUBTITLE,
+} from '../../../../../common/visualizations/lens/dashboards/host/kpi_grid_config';
+
+interface Props {
+  dataView?: DataView;
+  nodeName: string;
+  timeRange: TimeRange;
+}
+
+export const KPIGrid = React.memo(({ nodeName, dataView, timeRange }: Props) => {
```

Comment:
Even better will be when we manage to use this grid in the hosts view. Currently, the same grid is built here and there. It would be nice if the hosts view could use a KPIGrid component and not worry about its implementation details